### PR TITLE
Pepper 1040 unable obtain tissue fix

### PR DESCRIPTION
**Pull request by ssettipalli**
PEPPER-1040
set the tissue request : status/value to "UnableToObtain" when the "unable to obtain tissue" checkbox is checked/true

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/BasePatch.java [hunk 1]
```diff
@@ -279,17 +281,24 @@ protected List<NameValue> setWorkflowRelatedFields(@NonNull Patch patch) {
...
+                    nameValues.add(setAdditionalValue("oD.request",
+                            new Patch(patch.getId(), PARTICIPANT_ID, patch.getParentId(), patch.getUser(), patch.getNameValue(),
+                                    patch.getNameValues(), patch.getDdpParticipantId()), "sent"));
+                }
             } else {
+                //"unable to obtain tissue" : checked/true .. update request status
                 nameValues.add(setAdditionalValue("oD.request",
                         new Patch(patch.getId(), PARTICIPANT_ID, patch.getParentId(), patch.getUser(), patch.getNameValue(),
-                                patch.getNameValues(), patch.getDdpParticipantId()), "sent"));
+                                patch.getNameValues(), patch.getDdpParticipantId()), OncHistoryDetail.UNABLE_OBTAIN_TISSUE));
```
Comment (ssettipalli) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/BasePatch.java]: set the request : status/value when the "unable to obtain tissue"  checkbox is checked/true

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/generate/OncHistoryDetailUnableObtainTissueStrategy.java [hunk 2]
```diff
@@ -11,7 +16,9 @@ public OncHistoryDetailUnableObtainTissueStrategy(GeneratorPayload generatorPayl
     @Override
     public Map<String, Object> generate() {
         if (isUnableToObtain()) {
-            return Map.of();
+            Map<String, Object> resultMap = new HashMap<>();
```
Comment (ssettipalli) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/generate/OncHistoryDetailUnableObtainTissueStrategy.java:19]: to update in elastic

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/BasePatch.java [hunk 3]
```diff
@@ -279,17 +281,24 @@ protected List<NameValue> setWorkflowRelatedFields(@NonNull Patch patch) {
...
+            if (!(boolean) patch.getNameValue().getValue()) {
+                boolean hasReceivedDate = new OncHistoryDetailDaoImpl().hasReceivedDate(getOncHistoryDetailId(patch));
 
-            if (hasReceivedDate) {
-                nameValues.add(setAdditionalValue("oD.request",
-                        new Patch(patch.getId(), PARTICIPANT_ID, patch.getParentId(), patch.getUser(), patch.getNameValue(),
-                                patch.getNameValues(), patch.getDdpParticipantId()), "received"));
+                if (hasReceivedDate) {
+                    nameValues.add(setAdditionalValue("oD.request",
+                            new Patch(patch.getId(), PARTICIPANT_ID, patch.getParentId(), patch.getUser(), patch.getNameValue(),
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/BasePatch.java]: Perhaps simplify this code by setting the 'additionalValue' value in a conditional and making just one complex nested call. (I appreciate some of the code was already this way but this is a great opportunity to fix sub-par code and greatly improve readability.)

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/BasePatch.java [hunk 4]
```diff
@@ -279,18 +280,15 @@ protected List<NameValue> setWorkflowRelatedFields(@NonNull Patch patch) {
...
-                        new Patch(patch.getId(), PARTICIPANT_ID, patch.getParentId(), patch.getUser(), patch.getNameValue(),
-                                patch.getNameValues(), patch.getDdpParticipantId()), "received"));
-            } else {
-                nameValues.add(setAdditionalValue("oD.request",
-                        new Patch(patch.getId(), PARTICIPANT_ID, patch.getParentId(), patch.getUser(), patch.getNameValue(),
-                                patch.getNameValues(), patch.getDdpParticipantId()), "sent"));
-            }
+        } else if (patch.getNameValue().getName().equals("oD.unableObtainTissue")) {
+            String status = (boolean) patch.getNameValue().getValue()
+                    ? OncHistoryDetail.UNABLE_OBTAIN_TISSUE
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/BasePatch.java:287]: Improved, but this construct is hard to understand.
Perhaps an old school if statement would be easier to read?

**Review by denniscunningham — APPROVED** [review]
One comment.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/BasePatch.java (hunk 4 above)
Comment (ssettipalli) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/BasePatch.java:287]: added a comment which should help understanding it.

**Pull request merged**